pandas/core/indexes/multi.py

```diff
@@ -3486,10 +3488,12 @@ def intersection(self, other, sort=False):
         uniq_tuples = None  # flag whether _inner_indexer was successful
         if self.is_monotonic and other.is_monotonic:
             try:
-                uniq_tuples = self._inner_indexer(lvals, rvals)[0]
+                inner_tuples = self._inner_indexer(lvals, rvals)[0]
                 sort = False  # uniq_tuples is already sorted
```

Comment:
does this comment still make sense with your change?

Reply:
Yes, we create it in the else block